clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -175,8 +177,13 @@ protected abstract void onJoinComplete(int generation,
      */
     public void ensureCoordinatorReady() {
         while (coordinatorUnknown()) {
-            RequestFuture<Void> future = sendGroupCoordinatorRequest();
-            client.poll(future);
+            if (findCoordinatorFuture == null)
+                findCoordinatorFuture = sendGroupCoordinatorRequest();
+
+            client.poll(findCoordinatorFuture);
+
+            RequestFuture<Void> future = findCoordinatorFuture;
+            findCoordinatorFuture = null;
```

Comment:
Given that we always set the future to `null` here, is it correct that the reason that we do all of this logic is to potentially use a future initialised via `lookupCoordinator`? And is there a point in actually setting the `findCoordinatorFuture` in this method? If I understand the aim correctly, it could be clearer if we did:

```java
RequestFuture<Void> future = findCoordinatorFuture;
findCoordinatorFuture = null;
if (future == null)
    future = sendGroupCoordinatorRequest();
```

But I may be missing an important detail. :)